examples/jsm/utils/BufferGeometryUtils.js

```diff
@@ -1199,6 +1199,112 @@ function mergeGroups( geometry ) {
 
 }
 
+
+// Creates a new, non-indexed geometry with smooth normals everywhere except faces that meet at
+// an angle greater than the crease angle.
+function generateSmoothNormalsWithCrease( geometry, creaseAngle = Math.PI / 3 /* 60 degrees */ ) {
```

Comment:
To me the name feels a bit out of sync with other similar BufferGeometry and BufferGeometryUtils methods:

- .computeTangents 
- .computeMikkTSpaceTangents
- .computeVertexNormals

What do you think of a name like `.computeCreaseVertexNormals`, as well as modifying and returning the input geometry at the end for consistency?

Reply:
Yes I was trying to think of what to call it. All of the compute* functions modify the geometry in place while this one must generate a new geometry since `toNonIndexed` is called which generates a new geometry to work with.

I'm seeing that all the merge* and to* functions generate a new geometry as a result, though. How about something like `toCreasedVertexNormalGeometry` or `toCreasedNormalGeometry`?

Reply:
`computeMikkTSpaceTangents` has the same limitation and works around it, would this be OK?

https://github.com/mrdoob/three.js/blob/06160137744f186ad968a95f1eed450e85f7be2f/examples/jsm/utils/BufferGeometryUtils.js#L68-L105

Reply:
Oh I see - I didn't realize that. I don't really love the ambiguity of sometimes returning a new geometry and sometimes not. It seems most clear to consistently return one or the other imo but I'm not sure how strong my opinions are here - may others have thoughts? @mrdoob @Mugen87?

Reply:
Ideally, the `compute*()` functions in `BufferGeometryUtils` should modify the geometry in place and return a new instance. In this way, they behave similar to the methods of `BufferGeometry` (like `computeBoundingBox()`, `computeBoundingSphere()` or `computeVertexNormals()`. 

`to*()` methods should methods return _something new_. A new instance or a target object provided as a parameter.

Reply:
> I don't really love the ambiguity of sometimes returning a new geometry and sometimes not.

None of our `computeFoo` methods return a new geometry — note the last few lines of `computeMikkTSpaceTangents` ... the new geometry is just used internally while building.


```js
 if ( geometry !== _geometry ) { 
  
 	geometry.copy( _geometry ) 
  
 } 
  
 return geometry; 
```

Reply:
> None of our computeFoo methods return a new geometry

Oh I see. I'm happy to change it to that - it just seems a bit odd considering geometry buffer lengths are more or less considered immutable now. If a geometry has been rendered and then one of these functions has been called on an index geometry then I believe `dispose` should be called on it before rendering again, no?

Reply:
Hm I'm not sure about that. @Mugen87 what do you think? 

If that's the case then perhaps these methods should be `toCreasedNormals()` and `toMikkTSpaceTangents()` instead.

Reply:
The renderer assumes that buffer attributes do not change in size after their first usage.

If a change in size is necessary, `BufferGeometry.dispose()` has to be called and a new geometry created.

Reply:
Ok, how about this naming convention for methods returning a new geometry?

- toNonIndexed()
- toMikkTSpaceTangents( invert = true )

There are a few names that "crease angle normals" could reasonably be called, I think:

- toCreasedNormals( creaseAngle = Math.PI / 3 )
- toSplitNormals( creaseAngle = Math.PI / 3 )
- toFaceNormals( creaseAngle = Infinity )
- toSmoothNormals( creaseAngle = 0 )

I get much better search results for "split normals" than "creased normals", so that might be my vote.

Reply:
"Creased Normals" sounds a lot more descriptive and intuitive to me.

"Split" normals sounds and (from the blender results) looks like a geometry operation that results in this kind of creased look but it's not something I would think to look for without already knowing it existed. It definitely feels like a name that only really makes sense if you understand the algorithm or geometry restrictions in place that necessitate splitting vertices up in order to get this look.

Reply:
Sorry for the delay — so perhaps these names?

- `BufferGeometry.toNonIndexed()`
- `BufferGeometryUtils.toMikkTSpaceTangents( MikkTSpace, negateSign = true )`
- `BufferGeometryUtils.toCreasedNormals( creaseAngle = Math.PI / 3 )`

I'm happy with those if so.